src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,608 @@ const ActorInit En_Box_InitVars = {
+        EnBox_SetupAction(this, func_809C9700);
+        this->dyna.actor.flags |= 0x2000000;
+        func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y - 50.0f;
+        this->alpha = 0;
+        this->dyna.actor.flags |= 0x10;
+    } else if ((this->type == ENBOX_TYPE_SWITCH_FLAG_BIG) && (!Flags_GetSwitch(globalCtx, (s32)this->switchFlag))) {
+        EnBox_SetupAction(this, EnBox_AppearOnSwitchFlag);
+        func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y - 50.0f;
+        this->alpha = 0;
+        this->dyna.actor.flags |= 0x10;
+    } else {
+        if ((this->type == ENBOX_TYPE_4) || (this->type == ENBOX_TYPE_6)) {
+            this->dyna.actor.flags |= 0x80;
+        }
+        EnBox_SetupAction(this, EnBox_WaitOpen);
+        this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+        this->movementFlags |= ENBOX_MOVE_STICK_TO_GROUND;
+    }
+
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.posRot.rot.y += 0x8000;
+    this->dyna.actor.posRot.rot.z = this->dyna.actor.shape.rot.z;
+    this->dyna.actor.initPosRot.rot.z = this->dyna.actor.shape.rot.z;
+
+    SkelAnime_Init(globalCtx, &this->skelanime, &D_060047D8, animHeader, this->limbDrawTable, this->transitionDrawTable,
+                   5);
+    SkelAnime_ChangeAnim(&this->skelanime, animHeader, 1.5f, animFrameStart, animFrameCount, 2, 0.0f);
+
+    switch (this->type) {
+        case ENBOX_TYPE_SMALL:
+        case ENBOX_TYPE_6:
+        case ENBOX_TYPE_ROOM_CLEAR_SMALL:
+        case ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL:
+            Actor_SetScale(&this->dyna.actor, 0.005f);
+            Actor_SetHeight(&this->dyna.actor, 20.0f);
+            break;
+        default:
+            Actor_SetScale(&this->dyna.actor, 0.01f);
+            Actor_SetHeight(&this->dyna.actor, 40.0f);
+    }
+}
+
+void EnBox_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBox* this = THIS;
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void EnBox_RandomMovement(EnBox* this, Vec3f* pos, Vec3f* velocity, Vec3f* accel) {
+    f32 randomRadius = Math_Rand_ZeroOne() * 25.0f;
+    s16 randomAngle = Math_Rand_ZeroOne() * 0x10000;
+
+    *pos = this->dyna.actor.posRot.pos;
+    pos->x += Math_Sins(randomAngle) * randomRadius;
+    pos->z += Math_Coss(randomAngle) * randomRadius;
+
+    velocity->y = 1.0f;
+    velocity->x = Math_Sins(randomAngle);
+    velocity->z = Math_Coss(randomAngle);
+
+    accel->x = 0.0f;
+    accel->y = 0.0f;
+    accel->z = 0.0f;
+}
+
+/**
+ * Spawns effects randomly around the chest
+ * Used when the chest hits the ground after falling (FALL types)
+ */
+void EnBox_SpawnLandingEffects(EnBox* this, GlobalContext* globalCtx) {
```

Comment:
this is for dust specifically, so you can change func names and docs accordingly.
might be worth naming the random movement function after effects or dust specifically